Traceback (most recent call last):
  File "/Users/ethan/opt/miniconda3/envs/pythonbook3/lib/python3.11/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/Users/ethan/opt/miniconda3/envs/pythonbook3/lib/python3.11/site-packages/nbclient/client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ethan/opt/miniconda3/envs/pythonbook3/lib/python3.11/site-packages/jupyter_core/utils/__init__.py", line 166, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ethan/opt/miniconda3/envs/pythonbook3/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/Users/ethan/opt/miniconda3/envs/pythonbook3/lib/python3.11/site-packages/nbclient/client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "/Users/ethan/opt/miniconda3/envs/pythonbook3/lib/python3.11/site-packages/nbclient/client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/Users/ethan/opt/miniconda3/envs/pythonbook3/lib/python3.11/site-packages/nbclient/client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
from myst_nb import glue
from matplotlib import pyplot as plt
import seaborn as sns

fig = plt.figure()

#sns.set_context("notebook", font_scale=1.5)
sns.scatterplot(data = df,
                x = 'dan_sleep', 
                y = 'dan_grump')
fig.set(title = 'Grumpiness and sleep', ylabel = 'My grumpiness (0-100)', xlabel='My sleep (hours)')
sns.despine()

glue("sleepycorrelation-fig", fig, display=False)
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mAttributeError[0m                            Traceback (most recent call last)
Cell [0;32mIn[2], line 11[0m
[1;32m      7[0m [38;5;66;03m#sns.set_context("notebook", font_scale=1.5)[39;00m
[1;32m      8[0m sns[38;5;241m.[39mscatterplot(data [38;5;241m=[39m df,
[1;32m      9[0m                 x [38;5;241m=[39m [38;5;124m'[39m[38;5;124mdan_sleep[39m[38;5;124m'[39m, 
[1;32m     10[0m                 y [38;5;241m=[39m [38;5;124m'[39m[38;5;124mdan_grump[39m[38;5;124m'[39m)
[0;32m---> 11[0m [43mfig[49m[38;5;241;43m.[39;49m[43mset[49m[43m([49m[43mtitle[49m[43m [49m[38;5;241;43m=[39;49m[43m [49m[38;5;124;43m'[39;49m[38;5;124;43mGrumpiness and sleep[39;49m[38;5;124;43m'[39;49m[43m,[49m[43m [49m[43mylabel[49m[43m [49m[38;5;241;43m=[39;49m[43m [49m[38;5;124;43m'[39;49m[38;5;124;43mMy grumpiness (0-100)[39;49m[38;5;124;43m'[39;49m[43m,[49m[43m [49m[43mxlabel[49m[38;5;241;43m=[39;49m[38;5;124;43m'[39;49m[38;5;124;43mMy sleep (hours)[39;49m[38;5;124;43m'[39;49m[43m)[49m
[1;32m     12[0m sns[38;5;241m.[39mdespine()
[1;32m     14[0m glue([38;5;124m"[39m[38;5;124msleepycorrelation-fig[39m[38;5;124m"[39m, fig, display[38;5;241m=[39m[38;5;28;01mFalse[39;00m)

File [0;32m~/opt/miniconda3/envs/pythonbook3/lib/python3.11/site-packages/matplotlib/artist.py:147[0m, in [0;36mArtist.__init_subclass__.<locals>.<lambda>[0;34m(self, **kwargs)[0m
[1;32m    139[0m [38;5;28;01mif[39;00m [38;5;129;01mnot[39;00m [38;5;28mhasattr[39m([38;5;28mcls[39m[38;5;241m.[39mset, [38;5;124m'[39m[38;5;124m_autogenerated_signature[39m[38;5;124m'[39m):
[1;32m    140[0m     [38;5;66;03m# Don't overwrite cls.set if the subclass or one of its parents[39;00m
[1;32m    141[0m     [38;5;66;03m# has defined a set method set itself.[39;00m
[1;32m    142[0m     [38;5;66;03m# If there was no explicit definition, cls.set is inherited from[39;00m
[1;32m    143[0m     [38;5;66;03m# the hierarchy of auto-generated set methods, which hold the[39;00m
[1;32m    144[0m     [38;5;66;03m# flag _autogenerated_signature.[39;00m
[1;32m    145[0m     [38;5;28;01mreturn[39;00m
[0;32m--> 147[0m [38;5;28mcls[39m[38;5;241m.[39mset [38;5;241m=[39m [38;5;28;01mlambda[39;00m [38;5;28mself[39m, [38;5;241m*[39m[38;5;241m*[39mkwargs: [43mArtist[49m[38;5;241;43m.[39;49m[43mset[49m[43m([49m[38;5;28;43mself[39;49m[43m,[49m[43m [49m[38;5;241;43m*[39;49m[38;5;241;43m*[39;49m[43mkwargs[49m[43m)[49m
[1;32m    148[0m [38;5;28mcls[39m[38;5;241m.[39mset[38;5;241m.[39m[38;5;18m__name__[39m [38;5;241m=[39m [38;5;124m"[39m[38;5;124mset[39m[38;5;124m"[39m
[1;32m    149[0m [38;5;28mcls[39m[38;5;241m.[39mset[38;5;241m.[39m[38;5;18m__qualname__[39m [38;5;241m=[39m [38;5;124mf[39m[38;5;124m"[39m[38;5;132;01m{[39;00m[38;5;28mcls[39m[38;5;241m.[39m[38;5;18m__qualname__[39m[38;5;132;01m}[39;00m[38;5;124m.set[39m[38;5;124m"[39m

File [0;32m~/opt/miniconda3/envs/pythonbook3/lib/python3.11/site-packages/matplotlib/artist.py:1231[0m, in [0;36mArtist.set[0;34m(self, **kwargs)[0m
[1;32m   1227[0m [38;5;28;01mdef[39;00m [38;5;21mset[39m([38;5;28mself[39m, [38;5;241m*[39m[38;5;241m*[39mkwargs):
[1;32m   1228[0m     [38;5;66;03m# docstring and signature are auto-generated via[39;00m
[1;32m   1229[0m     [38;5;66;03m# Artist._update_set_signature_and_docstring() at the end of the[39;00m
[1;32m   1230[0m     [38;5;66;03m# module.[39;00m
[0;32m-> 1231[0m     [38;5;28;01mreturn[39;00m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_internal_update[49m[43m([49m[43mcbook[49m[38;5;241;43m.[39;49m[43mnormalize_kwargs[49m[43m([49m[43mkwargs[49m[43m,[49m[43m [49m[38;5;28;43mself[39;49m[43m)[49m[43m)[49m

File [0;32m~/opt/miniconda3/envs/pythonbook3/lib/python3.11/site-packages/matplotlib/artist.py:1223[0m, in [0;36mArtist._internal_update[0;34m(self, kwargs)[0m
[1;32m   1216[0m [38;5;28;01mdef[39;00m [38;5;21m_internal_update[39m([38;5;28mself[39m, kwargs):
[1;32m   1217[0m [38;5;250m    [39m[38;5;124;03m"""[39;00m
[1;32m   1218[0m [38;5;124;03m    Update artist properties without prenormalizing them, but generating[39;00m
[1;32m   1219[0m [38;5;124;03m    errors as if calling `set`.[39;00m
[1;32m   1220[0m 
[1;32m   1221[0m [38;5;124;03m    The lack of prenormalization is to maintain backcompatibility.[39;00m
[1;32m   1222[0m [38;5;124;03m    """[39;00m
[0;32m-> 1223[0m     [38;5;28;01mreturn[39;00m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_update_props[49m[43m([49m
[1;32m   1224[0m [43m        [49m[43mkwargs[49m[43m,[49m[43m [49m[38;5;124;43m"[39;49m[38;5;132;43;01m{cls.__name__}[39;49;00m[38;5;124;43m.set() got an unexpected keyword argument [39;49m[38;5;124;43m"[39;49m
[1;32m   1225[0m [43m        [49m[38;5;124;43m"[39;49m[38;5;132;43;01m{prop_name!r}[39;49;00m[38;5;124;43m"[39;49m[43m)[49m

File [0;32m~/opt/miniconda3/envs/pythonbook3/lib/python3.11/site-packages/matplotlib/artist.py:1197[0m, in [0;36mArtist._update_props[0;34m(self, props, errfmt)[0m
[1;32m   1195[0m             func [38;5;241m=[39m [38;5;28mgetattr[39m([38;5;28mself[39m, [38;5;124mf[39m[38;5;124m"[39m[38;5;124mset_[39m[38;5;132;01m{[39;00mk[38;5;132;01m}[39;00m[38;5;124m"[39m, [38;5;28;01mNone[39;00m)
[1;32m   1196[0m             [38;5;28;01mif[39;00m [38;5;129;01mnot[39;00m [38;5;28mcallable[39m(func):
[0;32m-> 1197[0m                 [38;5;28;01mraise[39;00m [38;5;167;01mAttributeError[39;00m(
[1;32m   1198[0m                     errfmt[38;5;241m.[39mformat([38;5;28mcls[39m[38;5;241m=[39m[38;5;28mtype[39m([38;5;28mself[39m), prop_name[38;5;241m=[39mk))
[1;32m   1199[0m             ret[38;5;241m.[39mappend(func(v))
[1;32m   1200[0m [38;5;28;01mif[39;00m ret:

[0;31mAttributeError[0m: Figure.set() got an unexpected keyword argument 'title'

